api/src/main/java/org/apache/iceberg/variants/VariantObject.java

```diff
@@ -57,4 +59,37 @@ static String asString(VariantObject object) {
 
     return builder.toString();
   }
+
+  static int hash(VariantObject self) {
+    int hash = 17;
+    for (String field : self.fieldNames()) {
```

Comment:
This has me a little concerned.  It appears that we'll introduce a lot of integer overflow since we're multiplying by the hash of the fields.